app/uk/gov/hmrc/hec/services/IFService.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.services
+
+import java.time.LocalDate
+
+import cats.data.EitherT
+import cats.implicits._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.http.Status.OK
+import play.api.libs.json.{Json, Reads}
+import uk.gov.hmrc.hec.connectors.IFConnector
+import uk.gov.hmrc.hec.models.ids.{CTUTR, SAUTR}
+import uk.gov.hmrc.hec.models.{AccountingPeriod, CTStatus, CTStatusResponse, Error, SAStatus, SAStatusResponse, TaxYear}
+import uk.gov.hmrc.hec.services.IFServiceImpl.{BackendError, DataError, IFError, RawCTSuccessResponse, RawFailureResponse, RawSASuccessResponse}
+import uk.gov.hmrc.hec.util.HttpResponseOps._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[IFServiceImpl])
+trait IFService {
+
+  def getSAStatus(utr: SAUTR, taxYear: TaxYear, correlationId: String)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, IFError, SAStatusResponse]
+
+  def getCTStatus(utr: CTUTR, startDate: LocalDate, endDate: LocalDate, correlationId: String)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, IFError, CTStatusResponse]
+
+}
+
+@Singleton
+class IFServiceImpl @Inject() (
+  IFConnector: IFConnector
+)(implicit ec: ExecutionContext)
+    extends IFService {
+
+  private def handleErrorPath(httpResponse: HttpResponse, utrType: String) = {
+    val responseError = s"Response to get $utrType status came back with status ${httpResponse.status}"
+    httpResponse.parseJSON[RawFailureResponse] match {
+      case Left(_)         => Left(BackendError(Error(s"$responseError; could not parse body")))
+      case Right(failures) =>
+        val errorMsg = s"$responseError - ${failures.failures}"
+        if (httpResponse.status === 404) {
```

Comment:
Not  a big deal, to maintain consistency we can use BAD_REQUEST instead of 404 like we are using OK on line 79.Please ignore if you have too much on your plate :)

Reply:
I would actually like to do this, but couldn't find the right import. It's a trait and it seems like it's mixed into the controller, but not accessible in the service. Any idea how to get hold of that import?

Reply:
i just wrote this in your code and it worked for me , :) no import needed.

Reply:
should be under import `play.api.http.Status.BAD_REQUEST`